File: src/main/java/tafqeetj/converters/BasicNumberConverterFactory.java
package tafqeetj.converters;

import tafqeetj.numbers.negative.NegativeBasicNumberPlacesNames;
import tafqeetj.numbers.positive.PositiveBasicNumbersPLacesNames;

public class BasicNumberConverterFactory {
    private static final BasicNumberConverter positiveConverter =
            new BasicNumberConverter(PositiveBasicNumbersPLacesNames.getInstance());
    private static final BasicNumberConverter negativeConverter =
            new BasicNumberConverter(NegativeBasicNumberPlacesNames.getInstance());

    private BasicNumberConverterFactory() {}

    public static BasicNumberConverter getPositiveConverter() {
        return BasicNumberConverterFactory.positiveConverter;
    }

    public static BasicNumberConverter getNegativeConverter() {
        return BasicNumberConverterFactory.negativeConverter;
    }
}

File: src/main/kotlin/tech/moussa/tafqeetk/converters/BasicNumberConverterFactory.kt
package tech.moussa.tafqeetk.converters

import tech.moussa.tafqeetk.numbers.negative.NegativeBasicNumberPlacesNames
import tech.moussa.tafqeetk.numbers.positive.PositiveBasicNumbersPlacesNames

object BasicNumberConverterFactory {
    val positiveConverter = BasicNumberConverter(PositiveBasicNumbersPlacesNames.getInstance())
    val negativeConverter = BasicNumberConverter(NegativeBasicNumberPlacesNames.getInstance())
}